### PR TITLE
Add service details card for object service dashboard

### PR DESCRIPTION
**Pull request by afreen23**
https://jira.coreos.com/browse/RHSTOR-319
https://projects.invisionapp.com/share/BRS4A5ETN3H#/screens/364134659
noobaa

### CODE REVIEW

File: frontend/packages/noobaa-storage-plugin/src/components/details-card/details-card.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+          <DetailItem
+            key="provider"
+            title="Provider"
+            value={getInfrastructurePlatform(infrastructureData)}
+            isLoading={!infrastructureLoaded}
+          />
+          <DetailItem
+            key="version"
+            title="Version"
+            value={getCephVersion(_.get(cephClusterData, '0'))}
```
Comment (spadgett) [inline frontend/packages/noobaa-storage-plugin/src/components/details-card/details-card.tsx]: I'm confused. If this is a `K8sResourceKind`, why are we getting `'0'`? Should this have been declared as an array?

File: frontend/packages/noobaa-storage-plugin/src/components/details-card/details-card.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+import {
+  DashboardCard,
+  DashboardCardBody,
+  DashboardCardHeader,
+  DashboardCardTitle,
+} from '@console/internal/components/dashboard/dashboard-card';
+import { DetailsBody, DetailItem } from '@console/internal/components/dashboard/details-card';
+import {
+  withDashboardResources,
+  DashboardItemProps,
```
Comment (spadgett) [inline frontend/packages/noobaa-storage-plugin/src/components/details-card/details-card.tsx]: alphabetize imports

File: frontend/packages/noobaa-storage-plugin/src/components/details-card/details-card.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+  _.get(cephCluster, 'spec.cephVersion.image');
+
+const NOOBAA_SYSTEM_NAME_QUERY = 'NooBaa_system_name';
+
+const cephClusterResource: FirehoseResource = {
+  kind: referenceForModel(CephClusterModel),
+  namespaced: true,
+  namespace: 'openshift-storage',
+  name: 'ceph-cluster',
+  isList: true,
```
Comment (spadgett) [inline frontend/packages/noobaa-storage-plugin/src/components/details-card/details-card.tsx]: This doesn't seem like it should be a list?

File: frontend/packages/noobaa-storage-plugin/src/plugin.ts [hunk 4]
```diff
@@ -1,10 +1,17 @@
 import * as _ from 'lodash';
 
-import { Plugin, ModelFeatureFlag, ModelDefinition } from '@console/plugin-sdk';
+import {
+  Plugin,
+  ModelFeatureFlag,
+  ModelDefinition,
+  DashboardsTab,
+  DashboardsCard,
```
Comment (spadgett) [inline frontend/packages/noobaa-storage-plugin/src/plugin.ts]: alpha

File: frontend/packages/noobaa-storage-plugin/src/components/details-card/details-card.tsx (hunk 1 above)
Comment (afreen23) [reply frontend/packages/noobaa-storage-plugin/src/components/details-card/details-card.tsx]: yes the returned ceph cluster resource is having actually `data: [{...}]` format .

File: frontend/packages/noobaa-storage-plugin/src/components/details-card/details-card.tsx (hunk 3 above)
Comment (afreen23) [reply frontend/packages/noobaa-storage-plugin/src/components/details-card/details-card.tsx]: the ceph cluster resource is being returned in an array of size one

Comment (afreen23) [reply frontend/packages/noobaa-storage-plugin/src/components/details-card/details-card.tsx]: @cloudbehl ^^

Comment (rawagner) [reply frontend/packages/noobaa-storage-plugin/src/components/details-card/details-card.tsx]: @afreen23 I tried on your env and the CRD's instance name is called `rook-ceph`, not `ceph-cluster`. Is that expected ? What is the difference ? Storage dashboard is looking for `ceph-cluster` https://github.com/openshift/console/pull/1803/files#diff-d1f681bbd4c42188c00ffd77a9b11d73R40

Comment (cloudbehl) [reply frontend/packages/noobaa-storage-plugin/src/components/details-card/details-card.tsx]: @rawagner The name of the storage cluster is not going to come from rook-operator and will not be rook-ceph. Right now for the reference, I have put ceph-cluster but I am not sure what will be the name once the OCS cluster is development ready. But yes if we are using rook operator we can change it to rook-ceph to the name.

The OCS operator is right now in the development phase and we don't have any CRD available for it. We need to change these few things once available.

Comment (rawagner) [reply frontend/packages/noobaa-storage-plugin/src/components/details-card/details-card.tsx]: ok, but what should noobaa use ? `ceph-cluster` or `rook-ceph` ? 

The Firehose resource definition is incorrect. It should either be

```suggestion
 const cephClusterResource: FirehoseResource = {
  kind: referenceForModel(CephClusterModel),
  namespaced: true,
  namespace: 'openshift-storage',
  name: 'ceph-cluster',
  isList: false,
  prop: 'ceph',
```
or 

```suggestion
 const cephClusterResource: FirehoseResource = {
  kind: referenceForModel(CephClusterModel),
  namespaced: true,
  namespace: 'openshift-storage',
  name: 'rook-ceph',
  isList: false,
  prop: 'ceph',
```

`isList` should be false

Comment (cloudbehl) [reply frontend/packages/noobaa-storage-plugin/src/components/details-card/details-card.tsx]: @rawagner I agree it should not use it. But We don't have a value to show inversion right now as it is going to be there from OCS operator CR's. The option is to put is as `-` right now or show some value which will then come from OCS operator. You can suggest us

@afreen23 my PR is already updated and tested with the changes @rawagner provided. https://github.com/openshift/console/pull/1803/files#diff-d1f681bbd4c42188c00ffd77a9b11d73R36

Comment (afreen23) [reply frontend/packages/noobaa-storage-plugin/src/components/details-card/details-card.tsx]: @cloudbehl @rawagner  I have marked it as false as well and it is showing - .

Comment (rawagner) [reply frontend/packages/noobaa-storage-plugin/src/components/details-card/details-card.tsx]: @afreen23 you need to change https://github.com/openshift/console/pull/1824/files#diff-e7956324aa3d22bd01ba7006f8a018feR93

`value={getCephVersion(_.get(cephClusterData, '0'))}` to `value={getCephVersion(cephClusterData)}`

Comment (afreen23) [reply frontend/packages/noobaa-storage-plugin/src/components/details-card/details-card.tsx]: @rwagner changed

**Pull request merged**